lib/hooks/useAuth.ts

```diff
@@ -150,7 +194,17 @@ export const useAuth = () => {
   };
 
   const login = async (
-    { username, password }: { username: string; password: string },
+    {
+      username,
+      password,
+      needsConfirmation,
+      didConfirm,
+    }: {
+      username: string;
+      password: string;
+      needsConfirmation: (s: boolean) => void;
```

Comment:
You can use the following type `React.Dispatch<React.SetStateAction<boolean>>` This way you don't need to pass in an arrow function and can pass the state setter directly

Reply:
Will do